### PR TITLE
feat: narrow hardcoded floor to FLOOR_TOOLS

### PR DESCRIPTION
**Pull request by dodizzle**
## Summary
- Replace the hardcoded 11-tool `--allowedTools` array in `agent/src/claude.ts` with a `FLOOR_TOOLS` constant containing only the 7 permanently non-revocable tools: Read, Write, Edit, Glob, Grep, TodoWrite, Skill.
- Bash, WebSearch, WebFetch, and Agent are no longer hardcoded — they now flow entirely through `liveClaudeConfig.allowedTools` / `extraAllowedTools`, sourced from the AgentTool DB table (seeded by DAT-1.2, backfilled by DAT-2.1).
- Update existing unit tests to assert the new 7-tool baseline, and add two new tests covering the empty-DB-rows case (no Bash/WebSearch/WebFetch/Agent) and the DB-populated case (Bash/Agent present exactly once).

## ⚠️ Deploy ordering
Per the task description: this must not go live before **DAT-1.2** (new-agent seeding) and **DAT-2.1** (existing-agent backfill) are both fully deployed, or every existing agent instantly loses Bash/WebSearch/WebFetch/Agent access. Current status: DAT-1.2 is merged and in the deploy pipeline (`deploying`), DAT-2.1 is `done`. Please confirm DAT-1.2 has finished deploying before merging/deploying this PR.

## Acceptance Criteria

| # | Criterion | Status |
|---|-----------|--------|
| 1 | claude.ts no longer hardcodes Bash/WebSearch/WebFetch/Agent; only 7 floor tools remain | MET |
| 2 | Existing tests updated from old 11-tool array to new 7-tool baseline | MET |
| 3 | New test: empty allowedTools → Bash/WebSearch/WebFetch/Agent absent, 7 floor tools present | MET |
| 4 | New test: allowedTools with Bash+Agent (DB) → present exactly once alongside floor tools | MET |

## Test Plan
- `bun test agent/src/claude.unit.test.ts` — 57/57 passing
- `tsc --noEmit` — clean
- `biome lint` — clean
- Coverage: 100% lines on `agent/src/claude.ts`
- [x] Pre-commit checks passing

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/agent.md [hunk 1]
```diff
@@ -122,6 +122,47 @@ There is no HTTP chat endpoint on the agent. Chat flows through the chat service
...
+**Tool changes flow:**
+1. Operator/integration updates AgentTool rows via the admin API (`POST`/`PATCH`/`DELETE /agents/:id/tools/:toolId`)
+2. Agent's 60-second config sync fetches updated `allowedTools` from `/agents/:id/config`
+3. Next Claude session (up to 1 minute later) sees the new tool list — no pod restart required
+
+### Example: narrowing an agent to read-only mode
+
+To disable shell access and external requests for a risky agent:
+
+1. Call `DELETE /agents/:id/tools/Bash` (via admin API)
```
Comment (dmcaulay) [inline docs/agent.md]: This route's `toolId` path param is the `AgentTool` row's cuid primary key (see `ToolIdParamSchema` in `admin/src/openapi-schemas.ts` and the `AgentTool` Prisma model), not the tool's `pattern` string. `DELETE /agents/:id/tools/Bash` won't match any row — the caller needs to `GET /agents/:id/tools` first, find the row whose `pattern` is `"Bash"`, and delete by that row's `id`. Same issue applies to the WebSearch/WebFetch lines below.

**Review by dmcaulay — COMMENTED** [review]
Verdict: COMMENT — Core change (narrowing the hardcoded floor to 7 tools, routing Bash/WebSearch/WebFetch/Agent through the DB-backed AgentTool table) is correct, safe (no undefined-crash risk), and well-tested — 57/57 in claude.unit.test.ts, all 4 acceptance criteria genuinely met. One doc-accuracy issue: the new "narrowing an agent to read-only mode" example in docs/agent.md gives `DELETE /agents/:id/tools/Bash` etc., but that route's `toolId` is the AgentTool row's cuid, not the tool name — following it as written won't work. Also: pr-title-lint is failing (title needs a Conventional Commits prefix per CLAUDE.md). Deploy-ordering note in the PR body is verified accurate against the task store (DAT-1.2 is still `deploying`, not yet `deployed`) and is structurally tracked via DAT-1.3's `dependencies` field.

**Review by dodizzle — COMMENTED** [review]
Verdict: APPROVE — Prior reviewer's toolId-semantics finding fixed and independently verified against admin/src/agent-tools.ts. Dedup/ordering logic unchanged, new tests meaningfully cover both acceptance-criteria edge cases, CI green, no bypass path for Bash/WebSearch/WebFetch/Agent outside the DB-backed AgentTool table.
